apps/dashboard/src/components/primitives/tag-input.tsx

```diff
@@ -39,47 +45,74 @@ const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
     setInputValue('');
   };
 
-  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
-    if (e.key === 'Enter') {
-      e.preventDefault();
-
-      if (inputValue === '') {
-        return;
-      }
-
-      addTag(inputValue);
-    }
-  };
-
   return (
-    <div className="flex flex-col space-y-2">
-      <InputField>
-        <Input
-          ref={ref}
-          type="text"
-          value={inputValue}
-          onKeyDown={handleKeyDown}
-          className={cn('flex-grow', className)}
-          placeholder="Type a tag and press Enter"
-          onChange={(e) => setInputValue(e.target.value)}
-          {...rest}
-        />
-      </InputField>
-      <div className="flex flex-wrap gap-2">
-        {tags.map((tag, index) => (
-          <Badge key={index} variant="outline" kind="tag">
-            {tag}
-            <button
-              type="button"
-              onClick={() => removeTag(tag)}
-              className="ml-1 rounded-full outline-none focus:outline-none"
-            >
-              <RiCloseFill className="size-3" />
-              <span className="sr-only">Remove tag</span>
-            </button>
-          </Badge>
-        ))}
-      </div>
+    <div>
+      <Popover open={isOpen}>
+        <Command>
+          <div className="flex flex-col gap-2">
+            <PopoverAnchor asChild>
+              <CommandInput
+                ref={ref}
+                autoComplete="off"
+                value={inputValue}
+                className={cn(inputVariants(), 'flex-grow', className)}
+                placeholder="Type a tag and press Enter"
+                onValueChange={(value) => {
+                  setInputValue(value);
+                  setIsOpen(true);
+                }}
+                onFocusCapture={() => setIsOpen(true)}
+                onBlurCapture={() => setIsOpen(false)}
+                {...rest}
+              />
+            </PopoverAnchor>
+            <div className="flex flex-wrap gap-2">
+              {tags.map((tag, index) => (
+                <Badge key={index} variant="outline" kind="tag" className="gap-1">
+                  <span>{tag}</span>
+                  <button type="button" onClick={() => removeTag(tag)}>
+                    <RiCloseFill className="size-3" />
+                    <span className="sr-only">Remove tag</span>
+                  </button>
+                </Badge>
+              ))}
+            </div>
+          </div>
+          <PopoverContent
+            className="p-1"
+            portal={false}
+            onOpenAutoFocus={(e) => {
+              e.preventDefault();
+            }}
+          >
+            <CommandList>
+              <CommandGroup>
+                {inputValue !== '' && (
+                  <CommandItem
+                    value={`input-${inputValue}`} // if this exists in the list below there are bugs introduces to the key up/down behaviour
```

Comment:
@desiprisg This comment doesn't make a lot of sense. Can you make it more descriptive?

Reply:
Done